src/narrativelog/routers/add_message.py

```diff
@@ -41,37 +41,57 @@ async def add_message(
     systems: None
     | list[str] = fastapi.Body(
         default=None,
-        description="Zero or more systems to which the message applies.",
+        description="Zero or more systems to which the message applies. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_json' instead.",
     ),
     subsystems: None
     | list[str] = fastapi.Body(
         default=None,
-        description="Zero or more subsystems to which the message applies",
+        description="Zero or more subsystems to which the message applies. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_json' instead.",
     ),
     cscs: None
     | list[str] = fastapi.Body(
         default=None,
         description="Zero or more CSCs to which the message applies. "
         "Each entry should be in the form 'name' or 'name:index', "
-        "where 'name' is the SAL component name and 'index' is the SAL index.",
+        "where 'name' is the SAL component name and 'index' is the SAL index. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_json' instead.",
     ),
     components: None
     | list[str] = fastapi.Body(
         default=None,
         description="Zero or more components to which the message applies. "
-        "Each entry should be a valid component name entry on the OBS jira project.",
+        "Each entry should be a valid component name entry on the OBS jira project. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_json' instead.",
     ),
     primary_software_components: None
     | list[str] = fastapi.Body(
         default=None,
         description="Primary software components to which the message applies. "
-        "Each entry should be a valid component name entry on the OBS jira project.",
+        "Each entry should be a valid component name entry on the OBS jira project. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_json' instead.",
     ),
     primary_hardware_components: None
     | list[str] = fastapi.Body(
         default=None,
         description="Primary hardware components to which the message applies. "
-        "Each entry should be a valid component name entry on the OBS jira project.",
+        "Each entry should be a valid component name entry on the OBS jira project. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_json' instead.",
+    ),
+    components_json: None
```

Comment:
Maybe the API documentation is in another place.  If this documentation that will show on swagger, there should be more detail here.  Its not clear to me if you are allowing for search against the parts of this JSONB file.  If you are allowing search against parts, it will take a special kind of search which means special kind of description for how to do it. When I've done something similar, I was able to make searches against the keys in the jsonb field to look like searches in the schema columns.  To do so, I confined the jsonb field to one level (keys within in are like column names in the schema, and the values associated with they keys are column values. )    I did that because end users (anyone in the world) might access the API.  There are more restrictions on your narrativelog API.  Or maybe you are handling it somewhere else and I just haven't gotten to that part yet.